### PR TITLE
Refactor Policy & Rule Form implementation

### PR DESCRIPTION
**Pull request by 3nvi**
## Background

This PR clears up a bit of complexity when it comes to the way `PolicyForm` and `RuleForm` are defined 

## Changes

- Remove `BaseRuleForm`
- Clear up complex typings
- Fix typing issues

## Testing

- Can create & edit policy
- Can create & edit rule

### CODE REVIEW

File: web/src/components/forms/BaseRuleForm/BaseRuleFormCoreFields/BaseRuleFormCoreFields.tsx [hunk 1]
```diff
@@ -116,7 +102,7 @@ const BaseRuleFormCoreFields: React.FC<BaseRuleCoreFieldsProps> = ({ type }) =>
           label="* ID"
           placeholder={`The unique ID of this ${type}`}
           name="id"
-          disabled={initialValues.id}
+          disabled={'id' in initialValues}
```
Comment (alexmylonas) [inline web/src/components/forms/BaseRuleForm/BaseRuleFormCoreFields/BaseRuleFormCoreFields.tsx:105]: nice

File: web/src/pages/EditRule/EditRule.tsx [hunk 2]
```diff
@@ -104,6 +79,29 @@ const EditRulePage: React.FC = () => {
     );
   }
 
+  // format any JSON returned from the server simply because we are going to display it
+  // within an online web editor. To do that we parse the JSON and re-stringify it using proper
+  // spacings that make it pretty (The server of course doesn't store these spacings when
+  // it stores JSON, that's why we are making those here in the front-end)
+  const { rule } = queryData;
+  const initialValues = {
```
Comment (3nvi) [inline web/src/pages/EditRule/EditRule.tsx:87]: added TS safety here. If we forget/mistype something TS will complain

File: web/src/pages/EditPolicy/EditPolicy.tsx [hunk 3]
```diff
@@ -106,6 +79,31 @@ const EditPolicyPage: React.FC = () => {
     );
   }
 
+  // format any JSON returned from the server simply because we are going to display it
+  // within an online web editor. To do that we parse the JSON and re-stringify it using proper
+  // spacings that make it pretty (The server of course doesn't store these spacings when
+  // it stores JSON, that's why we are making those here in the front-end)
+  const { policy } = queryData;
+  const initialValues = {
```
Comment (3nvi) [inline web/src/pages/EditPolicy/EditPolicy.tsx:87]: added TS safety here. If we forget/mistype something TS will complain

File: api/graphql/schema.graphql [hunk 4]
```diff
@@ -12,8 +12,8 @@ type Mutation {
   addDestination(input: DestinationInput!): Destination
   addComplianceIntegration(input: AddComplianceIntegrationInput!): ComplianceIntegration!
   addLogIntegration(input: AddLogIntegrationInput!): LogIntegration!
```
Comment (3nvi) [inline api/graphql/schema.graphql:14]: split `CreateOrModifyXXXInput` to `AddXXXInput` and `UpdateXXXInput`.

This allows us to have stronger typings

Comment (3nvi) [reply api/graphql/schema.graphql:14]: since these two types have different fields

**Comment by alexmylonas** [comment]
Can you double check creating a rule works? i couldn't manage to create one, also mage_test_ci fails

**Pull request merged**